stackscan.c

```diff
@@ -0,0 +1,147 @@
+#include "include/common.h"
+
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+void *csp;
+void *stack_top;
+
+void inspect_stack(void *stack)
+{
+	printf("offset: %lu\n", cheri_length_get(stack) - cheri_offset_get(stack));
+}
+
+bool is_stack_pointer(void *ptr)
+{
+	if (cheri_is_valid(ptr))
+	{
+		uint64_t base = cheri_base_get(stack_top);
+		uint64_t length = cheri_length_get(stack_top);
+		uint64_t address = cheri_address_get(ptr);
+
+		if (address >= base && address <= (base + length))
```

Comment:
()++

Reply:
???

Reply:
I find explicit evaluation order more appealing:  `( (address >= base) && ( address <= (base + length)) )`. 
I would also guard `base`, `length` and `address` with `const` if the intent is to use them for a single bounds check before they go out of scope, but that may be a bit much for this example.

Reply:
I am struggling to see what advantage these changes would give over the original code. [Which, admittedly, I would write as `address >= base && address <= base + length`, but that's nitpicking on my part.]

Reply:
It lowers the cognitive and recall load on (one of) the reviewers, leading to a quicker turn-around time as I won't have to double check operator precedence table. Explicit grouping parentheses also mean i can make use of my code editor to "highlight matching brace". For me personally, the less I have to do in my head, the better.

Reply:
I think it's a personal coding preference, which is absolutely fine, but I don't think we should push our personal coding preferences onto others in PRs, otherwise we'll never reach a fixed point.

Reply:
I agree that we're still calibrating our comments and that's a perfectly reasonable thing for us to be doing at the moment :)